loader/loader.go

```diff
@@ -427,12 +433,17 @@ func loadYamlModel(ctx context.Context, config types.ConfigDetails, opts *Option
 			}
 
 			if !opts.SkipValidation {
-				if err := schema.Validate(dict); err != nil {
-					return fmt.Errorf("validating %s: %w", file.Filename, err)
-				}
-				if _, ok := dict["version"]; ok {
-					opts.warnObsoleteVersion(file.Filename)
-					delete(dict, "version")
+				sch, ok := ctx.Value(consts.ComposeSchema{}).(schema.ComposeSchema)
+				if !ok {
+					return fmt.Errorf("could not validate yaml: %s", err)
+				} else {
+					if err := sch.Validate(dict); err != nil {
+						return fmt.Errorf("validating %s: %w", file.Filename, err)
+					}
+					if _, ok := dict["version"]; ok {
+						opts.warnObsoleteVersion(file.Filename)
+						delete(dict, "version")
+					}
```

Comment:
Same this should not have change as we only switch the Json Schema library, changes  should only apply in the `schema.go` file, no?

Reply:
I stored the schema in the context to compile it only once. That's the only change that is reflected in more lines due to the go boiler plate to get it from the context 
```
sch, ok := ctx.Value(consts.ComposeSchema{}).(schema.ComposeSchema)
!ok {
  return fmt.Errorf("could not validate yaml: %s", err)
} else {
  ...
}
```